src/parquet/arrow/io.h

```diff
@@ -36,65 +37,25 @@ namespace arrow {
 // existing Arrow memory pool. This way we can direct all allocations to a
 // single place rather than tracking allocations in different locations (for
 // example: without utilizing parquet-cpp's default allocator)
-class PARQUET_EXPORT ParquetAllocator : public MemoryAllocator {
+class PARQUET_EXPORT ParquetAllocator : public MemoryPool {
```

Comment:
Is there a real need for this? Can't we just pass the plain MemoryPool to Parquet?